requires.py

```diff
@@ -46,6 +46,11 @@ def list_unit_data(self):
             print(unit['cluster_name'])
         '''
         for conv in self.conversations():
-            yield {'cluster_name': conv.get_remote('cluster_name'),
```

Comment:
I recommend something like this:

```python
cluster_name = conv.get_remote('cluster_name') or conv.get_remote('cluster-name') 
yield {
       'cluster-name': cluster_name, 
       'cluster_name': cluster_name
```

That way we can be generous with what we receive and send.

Reply:
At first I was in agreement, but there's another ugly case here to consider.  Some charms may have been working in spite of this issue, and fixing it in this way may cause breakage.

A concrete example I can share is using the current promulgated versions of graylog and elasticsearch together.  Elasticsearch has the default cluster-name value of "elasticsearch".  Due to issue #11, Graylog pulls this as None via interface:elasticsearch.  This fix repairs that issue, but would have the side effect of updating /var/snap/graylog/common/server.conf and changing elasticsearch_cluster_name value from the incorrect blank value to the correct "elasticsearch" value - however, if Graylog was previously working in spite of that, it may break after the change has been done.  (I should test this; I haven't done so yet.)

The most backwards-compatible change I can come up with is to simply share the cluster_name and cluster-name values over the relation "as-is", e.g.:

```python3
yield {
       'cluster-name': conv.get_remote('cluster-name'), 
       'cluster_name': conv.get_remote('cluster_name'),
# ...
```

This doesn't feel particularly nice or elegant, but it would make the interface layer merely expose these values, and leave the burden of selecting the appropriate value (or continuing to use the wrong one) up to the charm on the requires side.  It seems the least likely to cause unexpected breakage, in my opinion.

Thoughts or additional feedback?